### PR TITLE
add toll cost parameter

### PR DESCRIPTION
**Pull request by dzinad**

### CODE REVIEW

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteOptions.java
Comment (RingerJK) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteOptions.java:849]:
```suggestion
  @SerializedName("compute_toll_cost")
  @Nullable
  public abstract Boolean computeTollCost();
```

File: CHANGELOG.md
Comment (RingerJK) [inline CHANGELOG.md:7]: a changelog item is not needed as we keep these changes undocumented cc @zugaldia

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteOptions.java
Comment (RingerJK) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteOptions.java:846]: replace javadoc with comments to make API undocumented cc @zugaldia 
```suggestion
  /*
   * Whether to return calculated toll cost for the route.
   * If null, default is false.
   *
   * return true if the request includes toll cost
   */
```

**Review by RingerJK — COMMENTED** [review]
let's add `toll_costs` object to `DirectionsRoute` with the PR and extend tests for `RouteOptions` (`RouteOptionsTest`) and `DirectionsRoute`(`DirectionsRouteTest`)

**Comment by VysotskiVadim** [comment]
> How do I test that it works? I mean the integration part. @mapbox/navigation-android

I usually modify and run com.mapbox.samples.BasicDirections#main

File: services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/PaymentMethodsTest.java
Comment (VysotskiVadim) [inline services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/PaymentMethodsTest.java:43]: why do you compare separate fields? Isn't it easier to compare 2 objects?
```suggestion
        PaymentMethod etc = PaymentMethod.builder().small(555.6).build();
        PaymentMethod cash = PaymentMethod.builder().standard(777.0).build();
        PaymentMethods original = PaymentMethods.builder()
                .etc(etc)
                .cash(cash)
                .build();
        String json = original.toJson();
        PaymentMethods deserialized = PaymentMethods.fromJson(json);
        assertEquals(original, deserialized);
```

File: services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/RouteOptionsTest.java
Comment (VysotskiVadim) [inline services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/RouteOptionsTest.java:344]: Add a test to verify default value?

**Review by VysotskiVadim — APPROVED** [review]
LGTM

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/PaymentMethod.java
Comment (dzinad) [reply services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/PaymentMethod.java:36]: Oh, I thought it was mandatory.

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/PaymentMethod.java
Comment (abhishek1508) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/PaymentMethod.java:18]: This class list different sizes of vehicles accepted and how much it costs for each different vehicle size. Let's change the name to `CostPerVehicleSize` as the data it hosts is not a `PaymentMethod`

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/PaymentMethod.java
Comment (abhishek1508) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/PaymentMethod.java:33]:
```suggestion
  /*
   * Returns the toll cost for a small sized vehicle.
   *
   * return toll cost
   */
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/PaymentMethod.java
Comment (abhishek1508) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/PaymentMethod.java:42]:
```suggestion
  /*
   * Returns the toll cost for a standard sized vehicle.
   *
   * return toll cost
   */
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/PaymentMethod.java
Comment (abhishek1508) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/PaymentMethod.java:51]:
```suggestion
  /*
   * Returns the toll cost for a middle sized vehicle.
   *
   * return toll cost
   */
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/PaymentMethod.java
Comment (abhishek1508) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/PaymentMethod.java:139]:
```suggestion
     * Toll cost for a middle sized vehicle.
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/PaymentMethods.java
Comment (abhishek1508) [reply services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/PaymentMethods.java:45]: @dzinad Let's make sure we address this before you merge

Comment (dzinad) [reply services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/PaymentMethods.java:45]: Why? It's a json object in the response. `PaymentMethod` represents this object.

**Comment by VysotskiVadim** [comment]
@dzinad , please do not forget to add the new request parameter to `DirectionsService`, I forgot to remind you during review 🙂

**Comment by dzinad** [comment]
> @dzinad , please do not forget to add the new request parameter to DirectionsService, I forgot to remind you during review 🙂

Done, thanks!

File: services-directions/src/main/java/com/mapbox/api/directions/v5/DirectionsService.java
Comment (VysotskiVadim) [inline services-directions/src/main/java/com/mapbox/api/directions/v5/DirectionsService.java:185]: maybe ?
```suggestion
    @Field("compute_toll_cost") Boolean computeTollCost,
```
```suggestion
    @Query("compute_toll_cost") Boolean computeTollCost,
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/DirectionsRoute.java
Comment (RingerJK) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/DirectionsRoute.java:349]: not needed  
```suggestion
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/TollCost.java
Comment (RingerJK) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/TollCost.java:15]:
```suggestion
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/TollCost.java
Comment (RingerJK) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/TollCost.java:81]:
```suggestion
```

**Review by RingerJK — APPROVED** [review]

**Comment by zugaldia** [comment]
@RingerJK @dzinad What's next for this PR to land?

**Comment by RingerJK** [comment]
@LukasPaczos proposed not to merge the PR, as the API is not public and it is against what we decided (expose via mapbox-java public API only). @LukasPaczos could you correct me if I missed something

**Comment by dzinad** [comment]
> @RingerJK @dzinad What's next for this PR to land?

Also [there's a bug](https://github.com/mapbox/navigation-sdks/issues/1774#issuecomment-1187704909) that's being fixed, so I need that to land in order to test my code.

**Comment by LukasPaczos** [comment]
> @LukasPaczos proposed not to merge the PR, as the API is not public and it is against what we decided (expose via mapbox-java public API only)

That's correct. Since we are waiting for the bug to be resolved anyway, we can wait for the team's review on the API exposure guide. If anything changes in the interim, we can sync and decide separately.

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteOptions.java
Comment (LukasPaczos) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteOptions.java:880]:
```suggestion
   * Whether to return calculated toll cost for the route, if data is available.
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteOptions.java
Comment (LukasPaczos) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteOptions.java:2024]:
```suggestion
     * Whether to return calculated toll cost for the route, if data is available.
```

File: services-directions-models/src/test/resources/route_options_v5.json
Comment (LukasPaczos) [inline services-directions-models/src/test/resources/route_options_v5.json:39]: This also should be added to `/services-directions/src/test/resources/route_options_v5.json`.

**Review by LukasPaczos — COMMENTED** [review]
We're missing a `MapboxDirectionsTest` addition.

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/DirectionsRoute.java
Comment (korshaknn) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/DirectionsRoute.java:149]: @dzinad hey! Why have you used  `multi-line` comment instead of `Javadoc` comment? Previously we always used `Javadoc`, so new comments don't looks consistent with old on a customer side.

image

Comment (korshaknn) [reply services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/DirectionsRoute.java:149]: ah, looks like we don't need them to be documented

Comment (LukasPaczos) [reply services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/DirectionsRoute.java:149]: yes, this is expected

**Pull request merged**
